src/pat/modal/modal.js

```diff
@@ -767,68 +761,64 @@ export default Base.extend({
 
     activateFocusTrap: function () {
         var self = this;
-        var inputsBody = self.$modal
-            .find("." + self.options.templateOptions.classBodyName)
-            .first()
-            .find("select, input[type!=hidden], textarea, button, a");
-        var inputsFooter = self.$modal
-            .find("." + self.options.templateOptions.classFooterName)
-            .first()
-            .find("select, input[type!=hidden], textarea, button, a");
+        const modal_el = self.$modal[0];
+        var inputsBody = modal_el
+            .querySelector(`.${self.options.templateOptions.classBodyName}`)
+            .querySelectorAll(`select, input:not([type="hidden"]), textarea, button, a`);
+        var inputsFooter = modal_el
+            .querySelector(`.${self.options.templateOptions.classFooterName}`)
+            .querySelectorAll(`select, input:not([type="hidden"]), textarea, button, a`);
         var inputs = [];
-        for (var i = 0; i < inputsBody.length; i++) {
-            if ($(inputsBody[i]).is(":visible")) {
-                inputs.push(inputsBody[i]);
-            }
-        }
-        for (var j = 0; j < inputsFooter.length; j++) {
-            if ($(inputsFooter[j]).is(":visible")) {
-                inputs.push(inputsFooter[j]);
+
+        for (const el of [...inputsBody, ...inputsFooter]) {
+            const style = window.getComputedStyle(el);
+            if (style.display !== 'none' && style.visibility !== 'hidden' && style.opacity !== '0') {
```

Comment:
Instead of this check, consider using `is_visible` from `patternslib/src/core/dom`

Reply:
this might be interresting for checking visibility in patternslib: https://developer.mozilla.org/en-US/docs/Web/API/Element/checkVisibility

Reply:
Oh nice! I love how the web is evolving.